reference_data/management/commands/update_gencode.py

```diff
@@ -2,138 +2,208 @@
 import gzip
 import logging
 import os
-from datetime import datetime
 from tqdm import tqdm
 
-from django.core.management.base import BaseCommand
+from django.core.management.base import BaseCommand, CommandError
+from django.db.utils import DataError
 
-from reference_data.models import GENOME_VERSION_GRCh37
-from reference_data.models import GencodeRelease, GencodeGene, GencodeTranscript
+from reference_data.management.commands.utils.download_utils import download_file
+from reference_data.models import GeneInfo, TranscriptInfo, GENOME_VERSION_GRCh37, GENOME_VERSION_GRCh38
 
 logger = logging.getLogger(__name__)
 
-
-RELEASE_NUMBER = 19
-RELEASE_DATE = datetime(2013, 12, 1)
-GENOME_VERSION = GENOME_VERSION_GRCh37
+GENCODE_GTF_URL = "ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_{gencode_release}/gencode.v{gencode_release}.annotation.gtf.gz"
+GENCODE_LIFT37_GTF_URL = "ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_{gencode_release}/GRCh37_mapping/gencode.v{gencode_release}lift37.annotation.gtf.gz"
 
 
 class Command(BaseCommand):
+    help = "Loads the GRCh37 and/or GRCh38 versions of the Gencode GTF from a particular Gencode release"
 
     def add_arguments(self, parser):
-        parser.add_argument('gencode_file', nargs='?')
+        parser.add_argument('--reset', help="First drop any existing records from GeneInfo and TranscriptInfo", action="store_true")
+        parser.add_argument('--gencode-release', help="gencode release number (eg. 28)", type=int, required=True, choices=range(19, 29))
+        parser.add_argument('gencode_gtf_path', nargs="?", help="(optional) gencode GTF file path. If not specified, it will be downloaded.")
+        parser.add_argument('genome_version', nargs="?", help="gencode GTF file genome version", choices=[GENOME_VERSION_GRCh37, GENOME_VERSION_GRCh38])
 
     def handle(self, *args, **options):
-        gencode_file_path = options.get('gencode_file')
 
-        # load gencode file
-        if not (gencode_file_path and os.path.isfile(gencode_file_path)):
-            url = 'ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.annotation.gtf.gz'
-            logger.info("Downloading %s" % url)
+        update_gencode(
+            gencode_release=options['gencode_release'],
+            gencode_file_path=options.get('gencode_gtf_path'),
```

Comment:
not a big deal, but it would be nice if you used the same arg name in the command line options and function call